web/generic_scripting.js

```diff
@@ -0,0 +1,50 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AppOptions } from "./app_options.js";
+import { loadScript } from "pdfjs-lib";
+
+class GenericScripting {
+  constructor() {
+    this._scriptElement = null;
+    this._ready = loadScript(AppOptions.get("sandboxBundleSrc")).then(evt => {
+      this._scriptElement = evt?.target;
```

Comment:
What's the purpose of saving the script reference? The existence of a `<script>` element in the DOM is only significant between insertion and execution. After a script has loaded, there is not really a purpose for it any more.

The only situation where it could make a difference is if the script grabs a reference to the `<script>` tag (e.g. `document.currentScript` at the start of the script) and actively watches the node. That's not likely.

Reply:
> What's the purpose of saving the script reference?

Given that the `loadScript` function inserts the script into the DOM, see https://github.com/mozilla/pdf.js/blob/b194c820bfbf30734589bb37f7f632bcc7790352/src/display/display_utils.js#L531-L546 my (possibly flawed) thinking was that we should make sure to clean-up that up as well.

> After a script has loaded, there is not really a purpose for it any more.

Unless I'm completely misunderstanding you, are you actually suggesting that we can simply remove the script from the DOM already in the `onload` handler?

Reply:
> Unless I'm completely misunderstanding you, are you actually suggesting that we can simply remove the script from the DOM already in the `onload` handler?

Yep exactly. The script has run, it does not serve any other purpose after that, other than the potential case that I described before, or to help those who inspect the DOM to see that the script has been inserted. In other words, removing it from `onload` in `loadScript` should be fine.

Reply:
Thanks for your help!

I've updated the PR to address this, however not unconditionally since I didn't fancy going through all existing `loadScript` call-sites.